File: algorithms/sorting/cpp/SelectionSort.cpp
#include <vector>
using namespace std;

class Sorter {
private:
  vector<int> v;

public:
  Sorter(vector<int> &v) : v(v) {}

  void selectionSort(vector<int> &v) {
    int n = v.size();

    for (int i = 0; i < n - 1; i++) {
      int minIndex = i;

      for (int j = i + 1; j < n; j++) {
        if (v[j] < v[minIndex]) {
          minIndex = j;
        }
      }

      if (minIndex != i) {
        swap(v[i], v[minIndex]);
      }
    }
  }
};

File: algorithms/sorting/go/selectionsort.go
package main

import "fmt"

func selectionSort(arr []int) {
	n := len(arr)

	for i := 0; i < n-1; i++ {
		minIndex := i

		for j := i + 1; j < n; j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}
		
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}
}

func main() {
	arr := []int{4, 1, 47, 64, 23, 6, 37,69}

	selectionSort(arr)
	fmt.Println("Sorted array:  ", arr)
}
